spark/src/main/java/org/apache/zeppelin/spark/PySparkInterpreter.java

```diff
@@ -411,10 +414,25 @@ public InterpreterResult interpret(String st, InterpreterContext context) {
     }
   }
 
+  public void interrupt() throws IOException {
+    if (pythonPid > -1) {
+      logger.info("Sending SIGINT signal to PID : " + pythonPid);
+      Runtime.getRuntime().exec("kill -SIGINT " + pythonPid);
+    } else {
+      logger.warn("Non UNIX/Linux system, close the interpreter");
```

Comment:
That's because creating process in windows creates a `pseudo process` (thread), and the `pid` is a negative number. ``